File: mobile/android/RoleplayAI/app/src/main/java/com/roleplayai/app/ai/AiEngine.kt
package com.roleplayai.app.ai

import kotlinx.coroutines.flow.Flow

data class GenerationSettings(
    val maxNewTokens: Int = 180,
    val temperature: Float = 0.8f,
    val topP: Float = 0.9f
)

interface AiEngine {
    val name: String
    fun generateStream(prompt: String, settings: GenerationSettings = GenerationSettings()): Flow<String>
}

File: mobile/ios/RoleplayAI/App/AI/AIEngine.swift
import Foundation

struct GenerationSettings: Codable, Hashable {
    var maxNewTokens: Int = 180
    var temperature: Double = 0.8
    var topP: Double = 0.9
}

protocol AIEngine {
    var name: String { get }
    func generateStream(prompt: String, settings: GenerationSettings) -> AsyncStream<String>
}

